functional-samples/tutorial.mole-game/mole/service-worker.js

```diff
@@ -0,0 +1,38 @@
+let failTimeout;
+let moleShowing = false;
+let controllerId;
+
+chrome.action.onClicked.addListener(() => {
+  if (!moleShowing) return;
+
+  chrome.runtime.sendMessage(controllerId, 'success');
+  hideMole();
+
+  failTimeout && clearTimeout(failTimeout);
+  failTimeout = undefined;
+});
+
+function showMole() {
+  chrome.action.setIcon({ path: 'icon-mole.png' });
+  moleShowing = true;
+}
+
+function hideMole() {
+  chrome.action.setIcon({ path: 'icon-empty.png' });
+  moleShowing = false;
+}
+
+chrome.runtime.onMessageExternal.addListener((msg) => {
+  controllerId = msg.id;
```

Comment:
Should we check whether the id is set?

Reply:
I'm fairly comfortable with this since we always set it. Extra error handling is nice but in this case I would lean towards matching the code shown in the video.